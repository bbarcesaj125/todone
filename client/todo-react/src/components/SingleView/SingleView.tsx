import React from "react";
import { useParams, useNavigate } from "react-router-dom";
import styled from "styled-components";

interface Props {
  items: Item[];
}
// Styles
const Article = styled.article``;
const Heading = styled.h2``;
const P = styled.p``;
const Button = styled.button`
  width: 100%;
  height: 50px;
  background-color: ${(props) => props.theme.switchBackground};
  color: ${(props) => props.theme.text};
  border: 1px solid ${(props) => props.theme.border};
  border-radius: 5px;
  cursor: pointer;
`;

const SingleView: React.FC<Props> = ({ items }) => {
  const { id } = useParams();

  const index = items.findIndex((object) => {
    return object.title === id;
  });

  const navigate = useNavigate();
  function handleClick() {
    navigate("/");
  }

  return (
    <Article>
      <Heading>{items[index].title}</Heading>
      <P>{items[index].description}</P>
      <br />
      <Button onClick={handleClick}>Go back</Button>
    </Article>
  );
};

export default SingleView;
