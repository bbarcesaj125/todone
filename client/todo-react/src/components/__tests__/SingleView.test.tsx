import React from "react";
import { render, screen } from "@testing-library/react";
import SingleView from "../SingleView";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import todoData from "../../todos.json";

const initialItems: Item[] = todoData;

/* const mockedUseNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"), // use actual for all non-hook parts
  useParams: () => ({
    id: "Github",
  }),
  useNavigate: () => mockedUseNavigate,
}));

it("renders correctly", () => {
  const tree = renderer
    .create(<SingleView items={initialItems}></SingleView>)
    .toJSON();
  expect(tree).toMatchSnapshot();
}); */

it("SingleView displays information about a single task", () => {
  const { asFragment } = render(
    <MemoryRouter initialEntries={["/task/Reading"]}>
      <Routes>
        <Route
          path="/task/:id"
          element={<SingleView items={initialItems}></SingleView>}
        />
      </Routes>
    </MemoryRouter>
  );

  const taskTitle = screen.getByText("Reading");
  expect(taskTitle).toBeInTheDocument;
  expect(asFragment()).toMatchSnapshot();
});
