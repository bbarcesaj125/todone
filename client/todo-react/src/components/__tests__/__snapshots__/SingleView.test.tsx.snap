// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleView displays information about a single task 1`] = `
<DocumentFragment>
  <article>
    <h1>
      Reading
    </h1>
    <p>
      Read a book
    </p>
    <br />
    <button>
      Go back
    </button>
  </article>
</DocumentFragment>
`;

exports[`SingleView should display information about a single task 1`] = `
<DocumentFragment>
  <article>
    <h1>
      Reading
    </h1>
    <p>
      Read a book
    </p>
    <br />
    <button>
      Go back
    </button>
  </article>
</DocumentFragment>
`;
